src/service-worker.ts

```diff
@@ -0,0 +1,79 @@
+/// <reference lib="webworker" />
+
+// Courtesy of https://dev.to/100lvlmaster/create-a-pwa-with-sveltekit-svelte-a36
+import { build, files, version } from '$service-worker';
+
+const worker = self as unknown as ServiceWorkerGlobalScope;
+const cacheName = `cache-${version}`;
+
+// `build` is an array of all the files generated by the bundler,
+// `files` is an array of everything in the `static` directory
+const toCache = build.concat(files);
+const staticAssets = new Set(toCache);
+
+worker.addEventListener('install', (event) => {
+  event.waitUntil(
+    caches
+      .open(cacheName)
+      .then((cache) => cache.addAll(toCache))
+      .then(() => worker.skipWaiting())
+  );
+});
+
+worker.addEventListener('activate', (event) => {
+  event.waitUntil(
+    caches.keys().then(async (keys) => {
+      // delete old caches
+      for (const key of keys) {
+        if (key !== cacheName) await caches.delete(key);
```

Comment:
issue (non-blocking): it's deleting those caches sequentially while it could delete them in parallel :)

Reply:
JavaScript (including workers) is single-threaded, so I wouldn't really call it _parallel_, but yeah, we don't know how the native functions are implemented under the hood, so I guess it wouldn't harm.